include/aws/common/json.h

```diff
@@ -0,0 +1,355 @@
+ * on the object/array containing the aws_json_node.
+ * @param number The number you want to store in the aws_json_node
+ * @return A new number aws_json_node
+ */
+struct aws_json_node *aws_json_make_node_number(double number);
+
+/**
+ * Creates a new array aws_json_node and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_node_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * Deleting this array will also delete any aws_json_nodes it contains.
+ * @return A new array aws_json_node
+ */
+struct aws_json_node *aws_json_make_node_array(void);
+
+/**
+ * Creates a new boolean aws_json_node with the given boolean and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_node_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * @param boolean The boolean you want to store in the aws_json_node
+ * @return A new boolean aws_json_node
+ */
+struct aws_json_node *aws_json_make_node_boolean(bool boolean);
+
+/**
+ * Creates a new null aws_json_node and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_node_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * @return
+ */
+struct aws_json_node *aws_json_make_node_null(void);
+
+/**
+ * Creates a new object aws_json_node and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_node_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * Deleting this object will also delete any aws_json_nodes it contains.
+ * @return
+ */
+struct aws_json_node *aws_json_make_node_object(void);
+// ====================
+
+// ====================
+// Value setters and getters
+
+/**
+ * Sets the string value of a string aws_json_node.
+ * @param node The string aws_json_node.
+ * @param string The new string value.
+ * @return True if the string was successfully set, otherwise false.
+ */
+bool aws_json_set_string(struct aws_json_node *node, char *string);
+
+/**
+ * Sets the number value of a number aws_json_node.
+ * @param node The number aws_json_node.
+ * @param number The new number value.
+ * @return True if the number was successfully set, otherwise false.
+ */
+bool aws_json_set_number(struct aws_json_node *node, double number);
+
+/**
+ * Sets the boolean value of a boolean aws_json_node.
+ * @param node The boolean aws_json_node.
+ * @param boolean The new boolean value.
+ * @return True if the boolean was successfully set, otherwise false.
+ */
+bool aws_json_set_boolean(struct aws_json_node *node, bool boolean);
+
+/**
+ * Gets the string of a string aws_json_node.
+ * @param node The string aws_json_node.
+ * @return The string of the string aws_json_node, otherwise NULL.
+ */
+char *aws_json_node_get_string(struct aws_json_node *node);
```

Comment:
[trivial] naming inconsistency between setters and getters. the setters have `_node_` in them but the getters don't.

the getter here is named better than the setter, since our naming conventions call for "classname_functionname" pattern

but we have something that's kiiiiind of like subclasssing, where boolean is a "subclass" of node so I could also buy into a name like:
`aws_json_boolean_get()` `aws_json_boolean_set()`

either way